src/main/scala/amba/axi4/Bundles.scala

```diff
@@ -70,6 +70,24 @@ class AXI4Bundle(params: AXI4BundleParameters) extends AXI4BundleBase(params)
   val b  = Irrevocable(new AXI4BundleB (params)).flip
   val ar = Irrevocable(new AXI4BundleAR(params))
   val r  = Irrevocable(new AXI4BundleR (params)).flip
+
+  def tieoff() {
+    ar.ready.dir match {
+      case INPUT =>
+        ar.ready := Bool(true)
+        aw.ready := Bool(true)
+        w.ready  := Bool(true)
+        r.valid  := Bool(false)
+        b.valid  := Bool(false)
+      case OUTPUT =>
+        ar.valid := Bool(false)
+        aw.valid := Bool(false)
+        w.valid  := Bool(false)
+        r.ready  := Bool(true)
+        b.ready  := Bool(true)
+      case _ =>
+    }
+  }
 }
```

Comment:
Here you've taken the opposite policy. You allowed requests to go through (ready=true) even though you never respond AR=>R, AW=>B. This seems the opposite of what you might want. At least AHB/APB actually make progress when you say 'ready true'; they are the zero device. Here, AXI is the request black-hold device.

Reply:
I think the blocking-forever policy is the right one. I'll make them consistent.